core/src/main/java/com/coderanch/blackjack/Card.java

```diff
@@ -69,6 +77,10 @@ Suit suit() {
         return this.suit;
     }
 
+    int[] values() {
+        return this.rank.points;
```

Comment:
You are returning the private array of a card's rank. This means that any other class can do something like this:

```java
queenOfHearts.values()[0] = 5;
```
Suddenly, all queens are worth 5 points instead of 10.

In general, the following advice applies:

- Don't accept arrays in non-private methods and constructors. Accept collections instead.
- When accepting a collection in a non-private method or constructor, make a defensive copy.
- Never return arrays. Always return collections.
- (Almost) never return modifiable collections. Make collections unmodifiable before you return them.

In this particular case though, I don't think that using a collection is the easiest solution. I would give every rank one fixed value (give aces the value 11) and add special handling for aces in the `Hand` class.